File: src/main/kotlin/com/oktfolio/kite/security/model/ExtendedUserDetails.java
package com.oktfolio.kite.security.model;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

/**
 * @author oktfolio oktfolio@gmail.com
 * @date 2020/04/27
 */
public class ExtendedUserDetails implements UserDetails {

    private Long id;
    private String username;
    private String mobile;
    private String email;
    private String password;
    private Boolean credentialExpired;
    private Boolean locked;
    private Boolean enabled;
    private Boolean expired;
    private Boolean deleted;
    private List<String> authorities;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return authorities.stream().map(SimpleGrantedAuthority::new).collect(Collectors.toSet());
    }

    @Override
    public String getPassword() {
        return this.password;
    }

    @Override
    public String getUsername() {
        return this.username;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }

    public void setAuthorities(List<String> authorities) {
        this.authorities = authorities;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getMobile() {
        return mobile;
    }

    public void setMobile(String mobile) {
        this.mobile = mobile;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public ExtendedUserDetails(Long id, String username, String mobile, String email, String password, List<String> authorities) {
        this.id = id;
        this.username = username;
        this.mobile = mobile;
        this.email = email;
        this.password = password;
        this.authorities = authorities;
    }
}

File: src/main/kotlin/com/oktfolio/kite/security/model/ExtendedUserDetails.kt
package com.oktfolio.kite.security.model

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

/**
 * @author oktfolio oktfolio@gmail.com
 * @date 2020/03/15
 */
class ExtendedUserDetails(
        var id: Long,
        private var username: String = "",
        var mobile: String? = "",
        var email: String? = "",
        private var password: String = "",
        private var authorities: List<String> = arrayListOf()) : UserDetails {

    override fun getAuthorities(): Collection<GrantedAuthority> {
        println(authorities.map { SimpleGrantedAuthority(it) })
        return authorities.map { SimpleGrantedAuthority(it) }
    }

    override fun isEnabled(): Boolean {
        return true
    }

    override fun getUsername(): String {
        return this.username
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun getPassword(): String {
        return this.password
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

}